frontend/src/metabase/query_builder/components/dataref/FieldList.tsx

```diff
@@ -19,32 +22,35 @@ interface FieldListProps {
 }
 
 const FieldList = ({ fields, onFieldClick }: FieldListProps) => (
-  <NodeListContainer>
-    <NodeListTitle>
-      <NodeListIcon name="table2" size="12" />
-      <NodeListTitleText>
-        {ngettext(
-          msgid`${fields.length} column`,
-          `${fields.length} columns`,
-          fields.length,
-        )}
-      </NodeListTitleText>
-    </NodeListTitle>
-    {fields.map(field => {
-      // field.icon() cannot be annotated to return IconName
-      // because metabase-lib cannot import from metabase.
-      const iconName = field.icon() as IconName;
-      const tooltip = iconName === "unknown" ? t`Unknown type` : null;
-      return (
-        <li key={field.getUniqueId()}>
-          <NodeListItemLink onClick={() => onFieldClick(field)}>
-            <NodeListItemIcon name={iconName} tooltip={tooltip} />
-            <NodeListItemName>{field.name}</NodeListItemName>
-          </NodeListItemLink>
-        </li>
-      );
-    })}
-  </NodeListContainer>
+  <DelayGroup>
+    <NodeListContainer>
+      <NodeListTitle>
+        <NodeListIcon name="table2" size="12" />
+        <NodeListTitleText>
+          {ngettext(
+            msgid`${fields.length} column`,
+            `${fields.length} columns`,
+            fields.length,
+          )}
+        </NodeListTitleText>
+      </NodeListTitle>
+      {fields.map(field => {
+        // field.icon() cannot be annotated to return IconName
+        // because metabase-lib cannot import from metabase.
+        const iconName = field.icon() as IconName;
+        const tooltip = iconName === "unknown" ? t`Unknown type` : null;
+        return (
+          <NodeListItem as="li" key={field.getUniqueId()}>
+            <NodeListItemLink onClick={() => onFieldClick(field)}>
+              <NodeListItemIcon name={iconName} tooltip={tooltip} />
+              <NodeListItemName>{field.name}</NodeListItemName>
+              <NodeListInfoIcon field={field} position="top-end" />
```

Comment:
IMO it'd be better if this popover opened on the left of the sidebar, [like this](https://github.com/metabase/metabase/assets/6830683/7944f3a9-fb13-4b15-959b-f1d684434146), so that it would not cover anything in the sidebar while user is interacting with the sidebar.

Reply:
That is not possible with the current popover from Mantine. @mazameli what do you think about this? I'm happy to implement this with more basic Mantine components, but want to make sure the design team is on board.

Reply:
It should not do that.